daemon/main.go

```diff
@@ -15,1098 +15,24 @@
-	if err != nil {
-		log.WithError(err).Fatalf("Failed to parse %s: %s",
-			option.MonitorAggregationName, err)
-	}
-	option.Config.Opts.SetValidated(option.MonitorAggregation, monitorAggregationLevel)
-
-	policy.SetPolicyEnabled(strings.ToLower(viper.GetString("enable-policy")))
-
-	if err := identity.AddUserDefinedNumericIdentitySet(fixedIdentity); err != nil {
-		log.Fatalf("Invalid fixed identities provided: %s", err)
-	}
-
-	if err := kvstore.Setup(kvStore, kvStoreOpts); err != nil {
-		addrkey := fmt.Sprintf("%s.address", kvStore)
-		addr := kvStoreOpts[addrkey]
-		log.WithError(err).WithFields(logrus.Fields{
-			"kvstore": kvStore,
-			"address": addr,
-		}).Fatal("Unable to setup kvstore")
-	}
-
-	if err := labels.ParseLabelPrefixCfg(validLabels, labelPrefixFile); err != nil {
-		log.WithError(err).Fatal("Unable to parse Label prefix configuration")
-	}
-
-	_, r, err := net.ParseCIDR(nat46prefix)
-	if err != nil {
-		log.WithError(err).WithField(logfields.V6Prefix, nat46prefix).Fatal("Invalid NAT46 prefix")
-	}
-
-	option.Config.NAT46Prefix = r
-
-	// If device has been specified, use it to derive better default
-	// allocation prefixes
-	if option.Config.Device != "undefined" {
-		node.InitDefaultPrefix(option.Config.Device)
-	}
-
-	if v6Address != "auto" {
-		if ip := net.ParseIP(v6Address); ip == nil {
-			log.WithField(logfields.IPAddr, v6Address).Fatal("Invalid IPv6 node address")
-		} else {
-			if !ip.IsGlobalUnicast() {
-				log.WithField(logfields.IPAddr, ip).Fatal("Invalid IPv6 node address: not a global unicast address")
-			}
-
-			node.SetIPv6(ip)
-		}
-	}
-
-	if v4Address != "auto" {
-		if ip := net.ParseIP(v4Address); ip == nil {
-			log.WithField(logfields.IPAddr, v4Address).Fatal("Invalid IPv4 node address")
-		} else {
-			node.SetExternalIPv4(ip)
-		}
-	}
-
-	// Read the service IDs of existing services from the BPF map and
-	// reserve them. This must be done *before* connecting to the
-	// Kubernetes apiserver and serving the API to ensure service IDs are
-	// not changing across restarts or that a new service could accidentally
-	// use an existing service ID.
-	if option.Config.RestoreState {
-		restoreServiceIDs()
-	}
-
-	k8s.Configure(k8sAPIServer, k8sKubeConfigPath)
-
-	// workaround for to use the values of the deprecated dockerEndpoint
-	// variable if it is set with a different value than defaults.
-	defaultDockerEndpoint := workloads.GetRuntimeDefaultOpt(workloads.Docker, "endpoint")
-	if defaultDockerEndpoint != dockerEndpoint {
-		containerRuntimesOpts[string(workloads.Docker)] = dockerEndpoint
-		log.Warn(`"docker" flag is deprecated.` +
-			`Please use "--container-runtime-endpoint=docker=` + defaultDockerEndpoint + `" instead`)
-	}
-
-	err = workloads.ParseConfigEndpoint(option.Config.Workloads, containerRuntimesOpts)
-	if err != nil {
-		log.WithError(err).Fatal("Unable to initialize policy container runtimes")
+	if strings.HasSuffix(os.Args[0], "cilium-agent") {
```

Comment:
Let's make this more resilient and use `path.Base()` and a string match. `HasSuffix()` can be a bit brittle.